File: WinForms/CS/MVVMMainDemo/CodeExamples.Modules/Navigation.cs
using DevExpress.DXperience.Demos.CodeDemo;

namespace DevExpress.MVVM.Demos.CodeExamples {
    public class NavigationModule : CodeTutorialControl {
        public NavigationModule()
            : base() {
        }
        protected override string PreprocessCodeBeforeCompilation(string code, ExampleLanguage language) {
            if(language == ExampleLanguage.Csharp) {
                code = code.Replace("typeof(MainViewModel)", "typeof(DXSample.SampleClass.MainViewModel)");
                code = code.Replace("typeof(AcceptFormViewModel)", "typeof(DXSample.SampleClass.AcceptFormViewModel)");
            }
            return code;
        }
        protected override string PreprocessCodeBeforeOpen(string code, ExampleLanguage language) {
            if(language == ExampleLanguage.Csharp)
                code = code.Replace("mainView.Parent = sampleHost", "mainView.Parent = this");
            return code;
        }
    }
}

File: WinForms/VB/MVVMMainDemo/CodeExamples.Modules/Navigation.vb
Imports DevExpress.DXperience.Demos.CodeDemo

Namespace DevExpress.MVVM.Demos.CodeExamples
    Public Class NavigationModule
        Inherits CodeTutorialControl
        Protected Overrides Function PreprocessCodeBeforeCompilation(code As String, language As ExampleLanguage) As String
            If language = ExampleLanguage.VB Then
                code = code.Replace("GetType(MainViewModel)", "GetType(DXSample.SampleClass.MainViewModel)")
                code = code.Replace("GetType(AcceptFormViewModel)", "GetType(DXSample.SampleClass.AcceptFormViewModel)")
                code = PatchBindCommandActionVB(code, "showBtn", "ShowAcceptDialog", "MainViewModel")
                code = PatchBindCommandActionVB(code, "acceptBtn", "Accept", "AcceptFormViewModel")
                code = PatchBindCommandActionVB(code, "showBtn", "ShowContact", "ContactsViewModel")
                code = PatchEventToCommandActionVB(code, "ShowModuleA", "NavigationViewModel")
                code = PatchEventToCommandActionVB(code, "ShowSomeModulesInDeferredManner", "NavigationViewModel")
                code = PatchEventToCommandActionVB(code, "OnLoad", "RootViewModel")
                If code.IndexOf("ContactsViewModel") > 0 Then
                    code = PatchSetBindingVB(code, "showBtn", "Function(btn) btn.Text", "SimpleButton, String", "Function(x) x.ShowContactTitle", "ContactsViewModel, String")
                End If
            End If
            Return code
        End Function
        Protected Overrides Function PreprocessCodeBeforeOpen(code As String, language As ExampleLanguage) As String
            If language = ExampleLanguage.VB Then
                code = code.Replace("mainView.Parent = sampleHost", "mainView.Parent = Me")
            End If
            Return code
        End Function
    End Class
End Namespace
